x-pack/test/api_integration/apis/security/api_keys.ts

```diff
@@ -12,347 +12,9 @@ import { FtrProviderContext } from '../../ftr_provider_context';
-          const updatedKey = queryResult.body.apiKeys.find(
-            (apiKey: { id: string }) => apiKey.id === id
-          );
-          expect(updatedKey).to.not.be(undefined);
-          expect(updatedKey.metadata).to.eql({ foo: 'bar' });
-          expect(updatedKey.role_descriptors?.cross_cluster?.indices).to.eql([
-            {
-              names: ['somethingelse*'],
-              privileges: ['read', 'read_cross_cluster', 'view_index_metadata'],
-              field_security: { grant: ['field3'] },
-              query: '{"bool":{"must_not":{"term":{"field2":"value3"}}}}',
-              allow_restricted_indices: false,
-            },
-          ]);
-        }
-      });
-
-      if (!basic) {
-        it(`Elasticsearch should reject an invalid cross cluster API Key configuration`, async () => {
-          const createResult = await supertest
-            .post('/internal/security/api_key')
-            .set('kbn-xsrf', 'xxx')
-            .send({
-              type: 'cross_cluster',
-              name: 'test_cc_api_key',
-              metadata: {},
-              access: {
-                search: [
-                  {
-                    names: ['logs*'],
-                  },
-                ],
-              },
-            });
-          expect(createResult.status).to.be(200);
-          const id = createResult.body.id;
-
-          await supertest
-            .put('/internal/security/api_key')
-            .set('kbn-xsrf', 'xxx')
-            .send({
-              type: 'cross_cluster',
-              id,
-              metadata: {
-                foo: 'bar',
-              },
-              access: {
-                search: [
-                  {
-                    names: ['logs*'],
-                    query: { bool: { must_not: { term: { field2: 'value2' } } } },
-                  },
-                ],
-                // replication section is not allowed if earch contains query or field_security
-                replication: {
-                  names: ['logs*'],
-                },
-              },
-            })
-            .expect(400);
-
-          await supertest
-            .put('/internal/security/api_key')
-            .set('kbn-xsrf', 'xxx')
-            .send({
-              type: 'cross_cluster',
-              id,
-              metadata: {
-                foo: 'bar',
-              },
-              access: {
-                search: [
-                  {
-                    names: ['logs*'],
-                    field_security: { grant: ['field2'] },
-                  },
-                ],
-                // replication section is not allowed if earch contains query or field_security
-                replication: {
-                  names: ['logs*'],
-                },
-              },
-            })
-            .expect(400);
-        });
-      }
-    });
-
+    // uses getServiceApiKeyPrivileges from the synthetics plugin
```

Comment:
I believe this is originally coming from https://github.com/elastic/kibana/pull/130918, @legrego do you remember why we need to depend on synthetics plugin here and cannot just duplicate the logic of `getServiceApiKeyPrivileges` in our tests?